### PR TITLE
Add extra params component to show fields specified in validation_sch…

### PR DESCRIPTION
**Pull request by jnation3406**
…ema of instrument_type that needs things populated in extra_params

Right now this just uses the `validation_schema` in the instrument_type in configdb to auto-generate UI elements for fields in the extra_params at the configuration or instrument_configuration level. In the future we may want to enhance this to support auto-generating from validation_schema at the generic mode level and configuration type level. I think this would go a long way to making the compose page generic enough for other observatories use without modification.

### CODE REVIEW

File: src/components/RequestGroupComposition/CustomField.vue [hunk 1]
```diff
@@ -23,9 +23,11 @@
           :value="value"
           :state="validationState"
           :type="type"
+          :list="field + '-field-' + $parent.id + '-list'"
```
Comment (jnation3406) [inline src/components/RequestGroupComposition/CustomField.vue:26]: The changes in this file are unrelated to the extra params fields, its just something I had locally to use in hermes UI at some point. It adds support for auto-completion lists for string input fields basically.

File: src/components/RequestGroupComposition/ExtraParamsFields.vue [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      // Clear the extra_params section and set any default values in the schema here
+      var cleanedExtraParams = {};
+      for (let fieldIndex in this.fields) {
+        let field = this.fields[fieldIndex];
+        if (field.default) {
+          cleanedExtraParams[field.field] = field.default;
+        }
+      }
+      // Add some other extra_params fields that are handled separately back into the config if they were there
+      let separate_params = ['offset_ra', 'offset_dec'];
```
Comment (mgdaily) [inline src/components/RequestGroupComposition/ExtraParamsFields.vue:118]: I wonder if we can specify these at a higher level, and pass them as a prop to keep this component more generic? Since offset_ra, offset_dec are only relevant at the InstrumentConfig level, right? So I could imagine this list growing at this level and conflating a lot of the different fields at various levels together. Does that make some sense?

File: src/components/RequestGroupComposition/ExtraParamsFields.vue [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      for (let index in separate_params) {
+        if (separate_params[index] in this.extraParams) {
+          cleanedExtraParams[separate_params[index]] = this.extraParams[separate_params[index]];
+        }
+      }
+      this.$emit('update:extraParams', cleanedExtraParams);
+      console.log('Schema changed!');
```
Comment (mgdaily) [inline src/components/RequestGroupComposition/ExtraParamsFields.vue:125]: Looks like a debug/testing log?

**Review by mgdaily — APPROVED** [review]
Again, impressed by how few code changes this took. I think this is the way forward. We should probably be using server driven UIs as much as possible for this very reason
